### DISCUSSION TITLE
Why Can't I understand Conflicts in Git vs Gihub?

### DISCUSSION THREAD

**Discussion by rdmccart** [community/community#108397] [GitHub Education]
### Hi everyone,

I'm again stymied by both git and github. I had a development branch (lets call it dev-a) that I cut from main branch. I committed my changes and merged to dev branch. A few weeks later I cut another branch from main, (lets call it release-a) and pulled origin dev-a. Gitbash told me I had two conflicts. That made sense to me because they were the files I worked on and committed to dev-a. I ran command git status and resolved the conflicts on the files, committed to release-a, and then pushed origin.

Next is what utterly confuses me. I went to github to make a pull request and I have other conflicts!!!! With 25 files that I never touched during my development!!!!

Why did git tell me I had conflicts, so I could fix them, just for github to tell me I have conflicts with other unrelated files!? Why didn't git tell me about these when I tried to merge or push my branch to origin or, frankly, at any time? How come the that files that github is telling me I have conflicts with are files I have never seen before nor ever touched in my life?

This sounds like conflicts that git or github are making and not me.

**Comment by spenserblack** [comment]
To get a better understanding on what happened, would you be able to provide the exact commands you did? For example, when you resolved the initial conflicts after the pulling dev-a into release-a, did you do something like this?
```bash
# resolve conflicts in your preferred editor
git add .
git commit -m "Resolve conflicts"
```

BTW the "GitHub Education" category isn't for getting educated about GitHub. [GitHub Education is its own thing](https://education.github.com/). I think "New to GitHub" or "Programming Help" are better-matching categories.
